app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFragment.kt

```diff
@@ -1,126 +1,389 @@
         analyticsController.logEvent(AnalyticsEvents.PROBLEMS_REFRESH)
     }
 
-    override fun onStart() {
-        super.onStart()
-        store.subscribe(this) { state ->
-            state.skipRepeats { oldState, newState ->
-                oldState.problems.status == newState.problems.status
-                        && oldState.problems.isFavourite == newState.problems.isFavourite
-                        && oldState.problems.filteredProblems == newState.problems.filteredProblems
-            }.select { it.problems }
+    private fun onFilter() {
+        startActivity(Intent(activity, ProblemsFiltersActivity::class.java))
+    }
+
+    private fun problemsRequest(query: String) {
+        store.dispatch(ProblemsRequests.SetQuery(query))
+    }
+}
+
+@Composable
+private fun ProblemsView(
+    problemsState: State<List<UIModel>>,
+    isRefreshState: SwipeRefreshState,
+    onProblem: (String, String) -> Unit,
+    onStar: (String) -> Unit,
+    onRefresh: () -> Unit,
+    onFilter: () -> Unit,
+    problemsRequest: (String) -> Unit
+) {
+    val swipeRefreshState = rememberSwipeRefreshState(isRefreshState.isRefreshing)
+
+    Box {
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(AlgoismeTheme.colors.primaryVariant)
+        )
+
+        Column {
+            NavigationBar(
+                onFilter = onFilter,
+                problemsRequest = problemsRequest
+            )
+
+            SwipeRefresh(
+                state = swipeRefreshState,
+                onRefresh = {
+                    onRefresh()
+                    isRefreshState.isRefreshing = true
+                }
+            ) {
+                ProblemsList(
+                    problemsState = problemsState,
+                    onProblem = onProblem,
+                    onStar = onStar
+                )
+            }
         }
     }
+}
 
-    override fun onStop() {
-        super.onStop()
-        store.unsubscribe(this)
+@Composable
+private fun ProblemsList(
+    problemsState: State<List<UIModel>>,
+    onProblem: (String, String) -> Unit,
+    onStar: (String) -> Unit,
+) = LazyColumn(
+    modifier = Modifier
+        .fillMaxSize()
+        .clip(RoundedCornerShape(30.dp, 30.dp, 0.dp, 0.dp))
+        .background(AlgoismeTheme.colors.primary)
+) {
+    items(problemsState.value) { problem ->
+        ProblemView(
+            problem = problem,
+            onProblem = onProblem,
+            onStar = onStar
+        )
     }
+}
 
-    override fun onNewState(state: ProblemsState) {
-        swipeRefreshLayout.isRefreshing = (state.status == ProblemsState.Status.PENDING)
-        problemsAdapter.setItems(state.filteredProblems, state.isFavourite)
+@Composable
+private fun ProblemView(
+    problem: UIModel,
+    onProblem: (String, String) -> Unit,
+    onStar: (String) -> Unit
+) = Row(
+    modifier = Modifier
+        .fillMaxWidth()
+        .padding(20.dp, 20.dp, 20.dp, 0.dp),
```

Comment:
It's better to add argument names for better readability (at least from GitHub).